src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -103,6 +109,102 @@ export class PatientViewPageStore {
         ))
     },{});
 
+    readonly geneticProfilesInStudy = remoteData(() => client.getAllGeneticProfilesInStudyUsingGET({
+        studyId: this.studyId
+    }), []);
+
+    private readonly mrnaRankGeneticProfileId = remoteData({
+        await: () => [
+            this.geneticProfilesInStudy
+        ],
+        invoke: () => {
+            const regex1 = /^.+rna_seq.*_zscores$/;
+            const regex2 = /^.*_zscores$/;
+            const chosenProfile: GeneticProfile = this.geneticProfilesInStudy.result.reduce((curr: GeneticProfile, next: GeneticProfile) => {
+                const nextId = next.geneticProfileId.toLowerCase();
+                if (curr && curr.geneticProfileId.toLowerCase().match(regex1) !== null) {

```

Comment:
regexp.test will return a boolean is and is more performance b/c it doesn't bother creating a match set